crates/rbuilder/src/live_builder/block_output/relay_submit.rs

```diff
@@ -274,11 +296,13 @@ async fn run_submit_to_relays_job(
             );
         }
 
-        if submission_optimistic {
-            let can_submit = if config.optimistic_prevalidate_optimistic_blocks {
+        if let Some(optimistic_signed_submission) = &optimistic_signed_submission {
+            let optimistic_config = optimistic_config.expect("optimistic config is not None");
```

Comment:
`expect` always makes me a bit nervous. it might hold true for now, but it's not guaranteed to hold in the future.

maybe ln 257 can return `Some((res, optimistic_config))`, which will bring the safely unwrapped config into scope here?

Reply:
That is a nice pattern :) Fixed!